### PR TITLE
Fix the test 'test_automated_recovery_from_stopped_node_and_start' to run with vsphere ipi

### PR DESCRIPTION
**Pull request by yitzhak12**
fix https://github.com/red-hat-storage/ocs-ci/issues/5278

### CODE REVIEW

File: ocs_ci/helpers/helpers.py [hunk 1]
```diff
@@ -3486,10 +3486,15 @@ def get_event_line_datetime(event_line):
          datetime object: The event line datetime
 
     """
-    if re.search(r"\d{4}-\d{2}-\d{2}", event_line):
-        return datetime.datetime.strptime(event_line[:26], "%Y-%m-%d %H:%M:%S.%f")
-    else:
-        return None
+    event_line_dt = None
+    if re.search(r"\d{4}-\d{2}-\d{2}T", event_line):
```
Comment (ebenahar) [inline ocs_ci/helpers/helpers.py:3490]: please assign `r"\d{4}-\d{2}-\d{2}` to a var and in here (line 3490) append `T`

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py [hunk 2]
```diff
@@ -219,12 +245,23 @@ def finalizer():
                     f"Successfully terminated node : "
                     f"{self.osd_worker_node[0].name} instance"
                 )
-            else:
+            elif get_node_status(self.osd_worker_node[0]) == constants.NODE_NOT_READY:
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:248]: get_node_status can return `None` as defined in https://github.com/red-hat-storage/ocs-ci/pull/5284/files#diff-1bba9ad83e489f5734ed993cec2d17018880017b2c2718e45607dcff3009f588R1917

I suggest you add a check the retuned value of `get_node_status()`

File: ocs_ci/ocs/node.py [hunk 3]
```diff
@@ -1898,3 +1898,20 @@ def get_odf_zone_count():
...
+    Args:
+        node_obj (ocs_ci.ocs.resources.ocs.OCS): The node object
+
+    Return:
+        str: The node status. If the command failed, it returns None.
+
+    """
+    try:
+        return node_obj.ocp.get_resource_status(node_obj.name)
+    except CommandFailed:
```
Comment (ebenahar) [inline ocs_ci/ocs/node.py:1916]: Why shouldn't we raise the exception if the node status failed to be fetched?

Comment (yitzhak12) [reply ocs_ci/ocs/node.py:1916]: Yes, I wasn't sure about it. I can remove the `try` and `exception` statements.

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py (hunk 2 above)
Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:248]: @ebenahar If we remove the `try` and `exception` statements as suggested in the comment below https://github.com/red-hat-storage/ocs-ci/pull/5284#discussion_r775810701, should we still need to check the return value of `get_node_status()`?

File: ocs_ci/helpers/helpers.py (hunk 1 above)
Comment (yitzhak12) [reply ocs_ci/helpers/helpers.py:3490]: Okay, that will do.

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py (hunk 2 above)
Comment (ebenahar) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:248]: I think it's better to check the returned value to be sure the status is returned correctly

Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py]: Fixed

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py [hunk 4]
```diff
@@ -203,12 +203,67 @@ def test_automated_recovery_from_failed_nodes_IPI_reactive(
...
 class TestAutomatedRecoveryFromStoppedNodes(ManageTest):
 
     osd_worker_node = None
     extra_node = False
+    machineset_name = None
+    start_ready_replica_count = None
+
+    def verify_osd_worker_node_in_ready_state(self, nodes):
+        """
+        Verify that the osd worker node is in ready state.
```
Comment (jilju) [inline tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:216]: This function is not just verifying whether the node status is Ready or not but trying to recover the node to Ready state. Please change the function name accordingly.

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py [hunk 5]
```diff
@@ -203,12 +203,67 @@ def test_automated_recovery_from_failed_nodes_IPI_reactive(
...
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=10,
+            func=machine.get_ready_replica_count,
+            machine_set=self.machineset_name,
+        )
+        try:
+            sample.wait_for_func_value(value=self.start_ready_replica_count)
+        except TimeoutExpiredError:
+            log.warning(
```
Comment (jilju) [inline tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:263]: raise Exception or
return False(and True for success).

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py (hunk 4 above)
Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:216]: Okay, that will do.

**Review by ebenahar — APPROVED** [review]

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py (hunk 5 above)
Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:263]: I see your point. I think the easiest way is to raise the exception in a case of failure.

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py [hunk 6]
```diff
@@ -203,12 +203,67 @@ def test_automated_recovery_from_failed_nodes_IPI_reactive(
...
+            f"Wait {timeout} seconds for the current ready replica count to be equal "
+            f"to the start ready replica count"
+        )
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=10,
+            func=machine.get_ready_replica_count,
+            machine_set=self.machineset_name,
+        )
+        sample.wait_for_func_value(value=self.start_ready_replica_count)
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:266]: These functions should not be inside the test file. Please take them our to a relevant utils or helpers file

Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py:266]: Yes, that will do.

**Review by ebenahar — APPROVED** [review]

**Review by annagitel — APPROVED** [review]
LGTM

**Pull request merged**